src/components/CatalogSearch/CatalogSearch.tsx

```diff
@@ -0,0 +1,112 @@
+import { Button, Grid, Input } from "@chakra-ui/react";
+import type {
+  ChangeEventHandler,
+  FormEventHandler,
+  FunctionComponent,
+} from "react";
+import {
+  Language,
+  TEMP_SUPPORTED_LANGUAGES,
+  LANGUAGE_NAME_MAP,
+} from "../../constants/languages";
+import { createTestIds } from "../../util/createTestIds";
+import { Dropdown, DropdownProps } from "./Dropdown";
+
+type LanguageItems = Partial<Record<Language, string>>;
+
+const languageOptions = Object.fromEntries(
+  Object.entries(LANGUAGE_NAME_MAP).filter(([key]) =>
+    TEMP_SUPPORTED_LANGUAGES.includes(key as Language)
+  )
+) as LanguageItems;
+
+const LanguageDropdown: FunctionComponent<DropdownProps<LanguageItems>> =
+  Dropdown;
+
+export const testIds = createTestIds("catalog-search", [
+  "form",
+  "input",
+  "languageDropdown",
+  "languageDropdownMenu",
+  "languageDropdownValue",
+  "languageItem",
+  "submit",
+] as const);
+
+export interface CatalogSearchProps {
+  /**
+   * Controls the query state value
+   */
+  query: string;
+  /**
+   * Controls the query state change event
+   */
+  onQueryChange: (search: string) => void;
+  /**
+   * Controls the language state value
+   */
+  language: Language | null;
+  /**
+   * Controls the language state change event
+   */
+  onLanguageChange: (language: Language | null) => void;
+  /**
+   * Called when the catalog search form is submitted (via enter keypress or submit click)
+   */
+  onSubmit: (params: {
+    query: CatalogSearchProps["query"];
+    language: CatalogSearchProps["language"];
+  }) => void;
+}
+
+export const CatalogSearch: FunctionComponent<CatalogSearchProps> = ({
+  query,
+  onQueryChange,
+  language,
+  onLanguageChange,
+  onSubmit,
+}) => {
+  const handleQueryChange: ChangeEventHandler<HTMLInputElement> = (e) => {
+    e.preventDefault();
+    onQueryChange(e.target.value);
+  };
+
+  const handleSubmit: FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+    onSubmit({ query, language });
+  };
```

Comment:
When accepting event handlers as props, I think we should just let the parent specify whether `preventDefault` is called or not and just accept the top level event handler function.

IE
```ts
<Input onChange={props.onChange} />
```
This makes the components more generic and allows callers more control over inputs.

Additionally, `onSubmit` accepts an object passing query and language, both of which are values controlled by the parent. There is no need for this component to know about the structure of this function and the values it accepts. Once again `onSubmit={props.onSubmit}` and the parent can pass the values where they need to go. This decouples the two significantly imo.